Sources/Algorithms/MinMax.swift

```diff
@@ -270,9 +270,9 @@ extension Collection {
       return []
     }
 
-    // If we're attempting to prefix more than 10% of the collection, it's
+    // If we're attempting to prefix more than log n of the collection, it's
     // faster to sort everything.
-    guard prefixCount < (self.count / 10) else {
+    guard prefixCount <= self.count._binaryLogarithm() else {
```

Comment:
You don't need underscored API for this:

```suggestion
    guard prefixCount <= (Int.bitWidth - self.count.leadingZeroBitCount) else {
```